core/client/src/main/java/alluxio/client/netty/NettyClient.java

```diff
@@ -37,8 +37,15 @@
   /**  Share both the encoder and decoder with all the client pipelines. */
   private static final RPCMessageEncoder ENCODER = new RPCMessageEncoder();
   private static final RPCMessageDecoder DECODER = new RPCMessageDecoder();
+  private static final boolean PACKET_STREAMING_ENABLED =
+      Configuration.getBoolean(PropertyKey.USER_PACKET_STREAMING_ENABLED);
 
-  private static final ChannelType CHANNEL_TYPE =
+  /**
+   * Packet streaming requires {@link io.netty.channel.epoll.EpollMode} to be set to
+   * LEVEL_TRIGGERED which is not supported in netty versions < 4.0.26.Final. Without shading
+   * netty in alluxio, we cannot use epoll.
+   */
+  private static final ChannelType CHANNEL_TYPE = PACKET_STREAMING_ENABLED ? ChannelType.NIO :
```

Comment:
Ideally the user can choose epoll even with packet streaming on since that is more ideal. I'll let you decide if it is worth the additional config value. If not, could you document that packet streaming always uses NIO?